### ISSUE TITLE
Unable to get GA Pod objects at runtime

### ISSUE THREAD

**Issue by sitefinitysteve** [NativeScript/ios-runtime#524]
So I'm basically trying to complete a GoogleAnalytics plugin, but on iOS at runtime I don't see any of the required objects.

I've seen this https://github.com/NativeScript/NativeScript/issues/1192 and there appears to be a modulemap file, but...well...I don't know :)

Thoughts?  I really need to get this up and running...

pod 'Google/Analytics'

(Apologies if this should be in the CLI or somewhere else, I went back and forth on it...)

**Comment by ivanbuhov** [comment]
Hi @sitefinitysteve 
The Google Analytics pod has a `module.modulemap` file but it is not working in the context of NativeScript because the pod declares its header folders as system header paths (more info [here](https://github.com/NativeScript/ios-runtime/issues/241#issuecomment-134557235)). A working solution is to add `module.modulemap` file in `{your-app}/platforms/ios/Pods/Headers/Public/` folder with the following content:
```
module Google2 {
  umbrella "."
  export *
}
```
This should do the trick. Unfortunately, this means that you have to add it manually every time you clone the repo, because it is not recommended to store `platforms` folder in the VCS.

**Comment by sitefinitysteve** [comment]
If shes a plugin though, could I not add it to <plugin>/platforms/ios/Pods/Headers/Public and have it all copied\merged?

**Comment by sitefinitysteve** [comment]
Oh, doesn't appear that stuff is copied over...?  Can we get it maybe copied and this would alleviate the problem?  Certainly it's not a viable option going forward to release a plugin that requires people to keep copying over a file everytime they flush the platform folder.

If you agree, should I close this, and re-open in the CLI repo?

**Comment by ivanbuhov** [comment]
It seems the pod command line tool is cleaning the `Pod/Headers` folder before `pod install` and the CLI can't do anything about it. I described the real solution [here](https://github.com/NativeScript/ios-runtime/issues/526) and it doesn't involve the CLI.

**Comment by ivanbuhov** [comment]
Hi @sitefinitysteve,
I came up with a working solution which doesn't require any manual work.
Create `ga-plugin` with the following structure:
![googleanalyticsnativescriptplugin](https://cloud.githubusercontent.com/assets/2405533/13664379/8eadde94-e6af-11e5-970f-b04615147a30.png)

The content of the `Podfile` is:
```
pod 'Google/Analytics'
```

The content of the `module.modulemap` file is:
```
module Google2 {
  umbrella "../../../../../platforms/ios/Pods/Headers/Public"
  export *
}
```

The content of `build.xcconfig` is:
```
HEADER_SEARCH_PATHS = $(inherited) $(PROJECT_DIR)/../../node_modules/ga-plugin/platforms/ios/headers
```

The tricky part here is that the `module.modulemap` file is never copied in the application's `platforms/ios` folder, but it exists in the `node_modules` folder. Also, the plugin includes `build.xcconfig` file which includes the path from `node_modules` in the list of header search path of the project.

**Comment by sitefinitysteve** [comment]
Seems to work! :D :+1: 

Thanks so much!

**Comment by sitefinitysteve** [comment]
@ivanbuhov Has something changed at all?  I've tried to load https://github.com/sitefinitysteve/nativescript-google-analytics on 3.2 and 3.3 but it can't ever find the GAI object, I've tried many variations of the modulemap but nothing seems to make the object available.  The Pod defiantly downloads, the .h files are there (and pretty sure I'm pointing to them properly)

So I loaded the plugin up yesterday to accept an android PR, tested, worked, loaded iOS to make sure it's okay, and all of a sudden it doens't work.  Noticed google changed the POD, so I updated that, API seems to be the same (awesome), but still no GAI avialable, the typings generator also doesn't see anything google related.
